src/Common/tests/System/Net/HttpTestServers.cs

```diff
@@ -15,12 +15,14 @@ internal class HttpTestServers
 
         public readonly static Uri RemoteGetServer = new Uri("http://" + Host + "/get");
         public readonly static Uri RemotePostServer = new Uri("http://" + Host + "/post");
+        public readonly static Uri RemotePutServer = new Uri("http://" + Host + "/put");
         public readonly static Uri SecureRemoteGetServer = new Uri("https://" + Host + "/get");
         public readonly static Uri SecureRemotePostServer = new Uri("https://" + Host + "/post");
         public readonly static Uri RemoteServerGzipUri = new Uri("http://" + Host + "/gzip");
         public readonly static Uri RemoteServerRedirectUri = new Uri("http://" + Host + "/relative-redirect/6");
         public readonly static Uri RemoteServerCookieUri = new Uri("http://" + Host + "/cookies");
         public readonly static Uri RemoteServerHeadersUri = new Uri("http://" + Host + "/headers");
+        public readonly static Uri SecureRemotePutServer = new Uri("https://" + Host + "/put");
```

Comment:
You should try to use [Theory]  and not [Fact] when doing this.  This will avoid repetitive test code.